metrics/server_config.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 Red Hat, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rs/zerolog/log"
+)
+
+type loggingResponseWriter struct {
+	http.ResponseWriter
+}
+
+func (writer loggingResponseWriter) WriteHeader(statusCode int) {
+	writer.ResponseWriter.WriteHeader(statusCode)
+	APIResponseStatusCodes.With(
+		prometheus.Labels{"status_code": fmt.Sprint(statusCode)},
+	).Inc()
+}
+
+func logRequestHandler(writer http.ResponseWriter, request *http.Request, nextHandler http.Handler) {
```

Comment:
why is it in metrics package? It's more like something related to a server.

Reply:
In fact it is related to both. It is a middleware for an HTTP server, but it simply do the "Inc" for the given metrics.

I think it is a good place, but probably bad file name... Let's see what is the opinion of the rest of the people :-)

Reply:
I'd move it to httputils package, but it's up to you :)

Reply:
Nice idea. Should we rename the `http` directory to `httputils`?